flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java

```diff
@@ -50,26 +50,26 @@ public void onReceive(final Context context, Intent intent) {
       notification.when = System.currentTimeMillis();
       NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
       notificationManager.notify(notificationId, notification);
-      boolean repeat = intent.getBooleanExtra("repeat", false);
-      if (!repeat) {
+      if (!intent.getBooleanExtra("repeat", false)) {
         FlutterLocalNotificationsPlugin.removeNotificationFromCache(context, notificationId);
       }
     } else {
-      Gson gson = FlutterLocalNotificationsPlugin.buildGson();
-      Type type = new TypeToken<NotificationDetails>() {}.getType();
+      final Gson gson = FlutterLocalNotificationsPlugin.buildGson();
+      final Type type = new TypeToken<NotificationDetails>() {}.getType();
       NotificationDetails notificationDetails = gson.fromJson(notificationDetailsJson, type);
-
-      if (notificationDetails.showNotification == null || notificationDetails.showNotification) {
+      final boolean locked = FlutterLocalNotificationsPlugin.isKeyguardLocked(context);
+      if (notificationDetails.showNotification == null
+          || notificationDetails.showNotification
+          || locked) {
         FlutterLocalNotificationsPlugin.showNotification(context, notificationDetails);
       } else {
         BackgroundAlarm.start(context, notificationDetails);
       }
       FlutterLocalNotificationsPlugin.scheduleNextNotification(context, notificationDetails);
 
-      boolean locked = FlutterLocalNotificationsPlugin.isKeyguardLocked(context);
-      boolean firstAlarm =
+      final boolean firstAlarm =
           LocalDateTime.parse(notificationDetails.scheduledDateTime).getSecond() == 0;
-      boolean hasStartActivity = notificationDetails.startActivityClassName != null;
+      final boolean hasStartActivity = notificationDetails.startActivityClassName != null;
```

Comment:
These changes are fine since this code does not exist in https://github.com/MaikuB/flutter_local_notifications/tree/master